client/settings/payment-settings/account-keys-connection-status.js

```diff
@@ -0,0 +1,232 @@
+`;
+
+export const AccountKeysConnectionStatus = ( { formRef } ) => {
+	const dispatch = useDispatch();
+	const {
+		isTesting,
+		isValid,
+		updateIsTestingAccountKeys,
+		updateIsValidAccountKeys,
+	} = useAccountKeys();
+
+	const handleTestConnection = async ( ref ) => {
+		updateIsTestingAccountKeys( true );
+		updateIsValidAccountKeys( null );
+
+		// Grab the HTMLCollection of elements of the HTML form, convert to array.
+		const elements = Array.from( ref.current.elements );
+		// Convert HTML elements array to an object acceptable for saving keys.
+		const keysToSave = elements.reduce( ( dict, ele ) => {
+			return Object.assign( dict, { [ ele.name ]: ele.value } );
+		}, {} );
+
+		let publishableKey;
+		let secretKey;
+
+		const isTestingLiveConnection =
+			keysToSave.publishable_key && keysToSave.secret_key;
+		const isTestingTestConnection =
+			keysToSave.test_publishable_key && keysToSave.test_secret_key;
+		if ( isTestingLiveConnection ) {
+			publishableKey = keysToSave.publishable_key;
+			secretKey = keysToSave.secret_key;
+			if (
+				! publishableKey.startsWith( 'pk_live_' ) ||
+				! secretKey.startsWith( 'sk_live_' )
```

Comment:
We must validate the use of restricted keys as well, starting with `rk_`, as mentioned in the input description.

Reply:
Good call. I didn't notice that.